ios/FluentUI/Core/Theme/Tokens/ShadowInfo.swift

```diff
@@ -59,4 +60,63 @@ public struct ShadowInfo: Equatable {
 
     /// The vertical offset of the shadow for shadow 2.
     public let yTwo: CGFloat
+
+    /// The number that the figma blur needs to be divided by to properly display shadows
+    private let blurDivisor: CGFloat = 2.0
+}
+
+public class ShadowUtil {
+
+    public static func applyShadow(_ shadowInfo: ShadowInfo, for view: UIView) {
+        guard let superview = view.superview else {
+            return
+        }
+        var shadowable: Shadowable
+
+        if view as? Shadowable != nil {
+            shadowable = view as! Shadowable
+        } else if view.superview as? Shadowable != nil {
+            shadowable = view.superview as! Shadowable
+        } else {
+            return
+        }
+
+        let shadowView = UIView()
+
+        shadowable.shadowView?.removeFromSuperview()
+        shadowable.shadowView = shadowView
+
+        shadowView.frame = view.frame
+        shadowView.layer.cornerRadius = view.layer.cornerRadius
+
+        superview.insertSubview(shadowView, belowSubview: view)
+
+        let shadow1 = CALayer()
+        let shadow2 = CALayer()
+
+        initializeShadowLayer(layer: shadow1, shadow: shadowInfo, view: view, isShadowOne: true)
+        initializeShadowLayer(layer: shadow2, shadow: shadowInfo, view: view)
+
+        shadowView.layer.insertSublayer(shadow1, at: 0)
+        shadowView.layer.insertSublayer(shadow2, below: shadow1)
+    }
+
+    private static func initializeShadowLayer(layer: CALayer,
+                                              shadow: ShadowInfo,
+                                              view: UIView,
+                                              isShadowOne: Bool = false) {
+        layer.frame = view.bounds
+        layer.shadowColor = UIColor(dynamicColor: isShadowOne ? shadow.colorOne : shadow.colorTwo).cgColor
+        layer.shadowRadius = isShadowOne ? shadow.blurOne : shadow.blurTwo
+        layer.shadowOpacity = 1
```

Comment:
Hmm in FURN we didn't do shadowOpacity = 1, we had to extract the opacity value from the alpha component of the color. The screenshots you have look fine though

Reply:
In UIKit we need to set the shadowOpacity to 1 since the alpha is already set in shadowColor.

Reply:
Might be worth leaving a comment explaining that, for future engineers who see this code.